userspace/h1b_tests/src/hil/flash/smart_program.rs

```diff
@@ -48,23 +48,22 @@ fn successful_program() -> bool {
 	hw.set_write_data(&[0xFFFF0FFF]);
 
 	// First attempt.
-	let mut state = h1b::hil::flash::smart_program::SmartProgramState::init(9);
-	state = state.step(&alarm, &hw, WRITE_OPCODE, /*is_timeout:*/ false);
-	require!(alarm.get_alarm() == alarm.now() + 150);
+	let mut state = h1b::hil::flash::smart_program::SmartProgramState::init(8, true, 100_000_000);
+	state = state.step(&alarm, &hw, WRITE_OPCODE);
+	require!(alarm.get_alarm() == alarm.now() + 1_600_000);
 	require!(hw.is_programming() == true);
 	require!(state.return_code() == None);
-	alarm.set_time(50);
 
-	// Inject a spurious interrupt.
-	state = state.step(&alarm, &hw, WRITE_OPCODE, /*is_timeout:*/ false);
-	require!(alarm.get_alarm() == 150);
+	// Finish.
+	alarm.set_time(1_600_000);
+	hw.inject_result(0);
+	state = state.step(&alarm, &hw, WRITE_OPCODE);
+	require!(alarm.get_alarm() == 3_200_000);
 	require!(hw.is_programming() == true);
 	require!(state.return_code() == None);
-	alarm.set_time(100);
+	alarm.set_time(3_200_000);
```

Comment:
Should this be set as a fraction of the frequency? Same question for similar cases below.

Reply:
These are now set in terms of MockAlarm's frequency. This is much more verbose -- do you still prefer it?